logic/src/commonMain/kotlin/com/wire/kalium/logic/data/user/UserMapper.kt

```diff
@@ -151,6 +154,27 @@ internal class UserMapperImpl(
         )
     }
 
+    override fun fromDaoModelToSelfUserWithTeam(userEntity: UserEntityWithTeam) = with(userEntity) {
+        val user = SelfUser(
+            id.toModel(),
+            name,
+            handle,
+            email,
+            phone,
+            accentId,
+            team?.let { TeamId(it) },
+            connectionStateMapper.fromDaoConnectionStateToUser(connectionState = connectionStatus),
+            previewAssetId?.toModel(),
+            completeAssetId?.toModel(),
+            availabilityStatusMapper.fromDaoAvailabilityStatusToModel(availabilityStatus)
+        )
+
+        val team = if (team != null && teamName != null && teamIcon != null) {
+            Team(team!!, teamName!!, teamIcon!!)
+        } else null
+        user to team
+    }
```

Comment:
This could be simplified if the DB already returned a pair, then we could just call a mapper for each element (`first` and `second`)